### PR TITLE
Integrate discount-codes API on tickets/discount-codes

### LINKED ISSUE #635: Implement discount code API on tickets/discount codes
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

List only , actions(edit, deactivate, delete) to be done in a separate issue.

### PR DESCRIPTION
**Pull request by CosmicCoder96**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Integrate discount-codes API on tickets/discount-codes
-  Add missing URL field to discount code model
- ~Add a temp fix to allow hasMany queries in ui-server~ Will be handled later

Fixes #635

### CODE REVIEW

**Comment by sumedh123** [comment]
@CosmicCoder96 Can you include a comment in ui-table-server that the changes are temporarily commented?

**Comment by niranjan94** [comment]
Deployment link: https://pr-643-open-event-frontend-fossasia.surge.sh

File: app/controllers/events/view/tickets/discount-codes/list.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      template     : 'components/ui-table/cell/events/view/tickets/discount-codes/cell-url',
+      title        : 'Discount code URL'
+    },
+    {
+      propertyName : 'value',
+      template     : 'components/ui-table/cell/events/view/tickets/discount-codes/cell-value',
+      title        : 'Discount Per Ticket'
+    },
+    {
+      propertyName : 'validTill',
```
Comment (sumedh123) [inline app/controllers/events/view/tickets/discount-codes/list.js]: Why did we not use here 'valid-till'?

File: app/controllers/events/view/tickets/discount-codes/list.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+const { Controller } = Ember;
+export default Controller.extend({
+  columns: [
+    {
+      propertyName : 'code',
+      title        : 'Discount code'
+    },
+    {
+      propertyName : 'discount-url',
```
Comment (geekyd) [inline app/controllers/events/view/tickets/discount-codes/list.js]: Disable sort for this column

**Comment by CosmicCoder96** [comment]
@geekyd @niranjan94 Made the changes

**Comment by niranjan94** [comment]
Deployment link: https://pr-643-open-event-frontend-fossasia.surge.sh

**Comment by niranjan94** [comment]
Deployment link: https://pr-643-open-event-frontend-fossasia.surge.sh

**Comment by niranjan94** [comment]
Deployment link: https://pr-643-open-event-frontend-fossasia.surge.sh

**Comment by sumedh123** [comment]
@CosmicCoder96 I am getting 500 on discount codes route on the deployment link. The server response is saying "Unprocessable entity".
```
{  
   "errors":[  
      {  
         "status":422,
         "source":{  
            "source":""
         },
         "title":"Unprocessable Entity",
         "detail":"Please check used_for and endpoint and verify your permission"
      }
   ],
   "jsonapi":{  
      "version":"1.0"
   }
}
```

**Comment by CosmicCoder96** [comment]
@sumedh123 Working for me 🤔 
image

**Comment by sumedh123** [comment]
@CosmicCoder96 I checked on the surge.sh link

**Comment by CosmicCoder96** [comment]
@sumedh123 me too

**Comment by CosmicCoder96** [comment]
@sumedh123 Figured it out, the discount codes requires organizer access, that's why you can't see it.

**Comment by sumedh123** [comment]
@CosmicCoder96 Okay.

**Pull request merged**